Find a working combination of packages

milljm opened the pull request:
This is a work in progress in an attempt to find a working compatible set of packages to satisfy paraview and moose-dev (seems to come down to hdf5)

Refs #24661

Commit:
Find a working combination of packages

Update packages to make Peacock and its dependencies coexist with
the bulk packages necessary to build/run MOOSE based applications.

Refs #24661

milljm commented:
I'll generate hashes for versioner when things appear to be working...

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1655117/) on 927dc24 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1655117/) on 927dc24 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1655117/) on 927dc24 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1655291/) on 6b95ea3 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1655291/) on 6b95ea3 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1655291/) on 6b95ea3 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1656160/) on 071ad84 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1656160/) on 071ad84 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1656207/) on bbab05f wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1656207/) on bbab05f wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1656207/) on bbab05f wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1656207/) on bbab05f wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1656207/) on bbab05f wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1656207/) on bbab05f wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1656723/) on 07eea16 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1656723/) on 07eea16 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1656723/) on 07eea16 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1656723/) on 07eea16 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1656723/) on 07eea16 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1656723/) on 07eea16 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1657268/) on 11b1a63 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1657268/) on 11b1a63 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1657268/) on 11b1a63 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
Dependency constraint adjustments

This allows us to drop moose-tools -> moose-mpich dependency I
thought we were going to need.

It is `clang-tools` which seems to spur a lot of conflicts. So by
NOT constraining tools and test-tools to a specific clang-tools
version, we can allow any clang-tools version to be install which is
supported by moose-mpich (but not _installed_ by moose-mpich).

While we're discusing... it is advisable to never place a version
constraint on any package in our tools and test-tools package...
Except of course, Python and the tools -> test-tools dependency.

milljm commented on conda/mpich/meta.yaml:
```diff
@@ -67,6 +61,12 @@ requirements:
+    - libtool
+    - autoconf
+    - automake 1.16.1
+    - m4
```
I just moved this around for visual appearance in the file.

cticenhour replied:
✨ 👀

milljm commented on conda/peacock/conda_build_config.yaml:
```diff
@@ -1,20 +1,5 @@
-moose_vtk:
-  - vtk 9.1.0
+moose_mpich:
+  - moose-mpich 4.1.1 build_0
```
Finally :)

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1657890/) on 4673cb6 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1657890/) on 4673cb6 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1657890/) on 4673cb6 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
libmesh-vtk to have a string build

I am setting precise dependencies in a downstream package, and it
is easier when we know what the string (hash) will be up front.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1659649/) on ab5f47e wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1659649/) on ab5f47e wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1659693/) on ab5f47e wanted to post the following:


## Framework coverage

Coverage did not change


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/24949/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/24949/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/24949/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/24949/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/24949/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/24949/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/24949/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/24949/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/24949/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/24949/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/24949/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/24949/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/24949/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/24949/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/24949/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/24949/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/24949/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/24949/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/24949/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/24949/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/24949/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/24949/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/24949/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/24949/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/24949/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/24949/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/24949/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/24949/coverage/xfem)

</details>



This comment will be updated on new commits.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1659675/) on ab5f47e wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/24949/site)

This comment will be updated on new commits.

Commit:
Skip tests which need IN_TREE, fix warnings

The following changes are to fix tests which cannot run in an
installed method.

Fix compiler warnings.

Closes #24661

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1666680/) on bd98974 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1666680/) on bd98974 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

lindsayad commented on modules/contact/test/tests/mortar_cartesian_lms/tests:
```diff
@@ -10,6 +10,7 @@
     max_parallel = 4
     min_ad_size = '50'
     valgrind = 'heavy'
+    installation_type = in_tree
```
Why is this needed?

milljm replied:
This test does not seem to work when ran out of tree. That is, `./configure --prefix /some/path; make install`

cticenhour replied:
@lindsayad any more comment here? This package update is getting close to being ready to merge - just turned on the full complement of `next` testing.

lindsayad replied:
I just have no idea why this test should be different than any other. But it's definitely not important enough to me to prevent this moving forward

milljm replied:
Good idea about the "full `next` complement".

milljm replied:
This test is fine.

I think what happened is that I mistook a faulty test for an actual MPI version issue (which I redacted several commits ago).

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1666931/) on d2959c5 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1666931/) on d2959c5 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1666931/) on d2959c5 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
Bump versions based on libMesh update

libMesh update.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1669971/) on b2c4d46 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
Constrain zfp

A new requirement during PETSc build (for superlu_dist). Need to
constrain zfp below 1.0.0.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1672097/) on 18c6aad wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1672097/) on 18c6aad wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1672097/) on 18c6aad wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
Perform clang-format

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1672254/) on 7efedf1 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1672254/) on 7efedf1 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1672254/) on 7efedf1 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

lindsayad commented:
Is this PR in a draft state?

milljm commented:
Its almost ready. Just this pesky Sandbox thing I've reported internal.

Commit:
Remove unused variables

This represents unused variables the new compiler picked up.

Commit:
Back off MPI version

MPICH 4.1.1 proved to "new". Caused countless issues, one with
sandboxing on Macintosh.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1682990/) on 1c19ea8 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1682990/) on 1c19ea8 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1682990/) on 1c19ea8 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
Fix conflicts on Linux

ICU and Autotools packages, appear to be the culprit somehow. Need to
pin these at specific versions. The bump to later MPICH requires
autotools bump. While doing so allowed a later version of ICU to
enter the mix, while all the peacock dependencies were still using
an older ICU package version. And because we have `strict` turned
on, moose-mpich would disallow the downgrade.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1694327/) on a143a56 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1694327/) on a143a56 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1694327/) on a143a56 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1698171/) on 8f187a3 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1698171/) on 8f187a3 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Conda build (ARM Mac)](https://civet.inl.gov/job/1698175/) on 8f187a3 : invalidated by @milljm

Commit:
Fix Intel Mac conflicts

Never ends.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1701089/) on a9e80e2 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1701089/) on a9e80e2 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1701089/) on a9e80e2 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
Do not bump compilers, or MPICH

Well, I give up bumping the toolchain while also finding a path
towards a working combination of compilers and Peacock environment.

Using the same compilers we've been using for a while, figure out
a working combination.

Commit:
Bump hashes

Still have work to do for Intel arch. Further bumps necessary.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1703292/) on 533688c wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1703292/) on 533688c wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1703292/) on 533688c wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
The changes needed for Intel Macs

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1703373/) on c1c5ce1 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1703373/) on c1c5ce1 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1703373/) on c1c5ce1 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
Bumping dates

With a working commit, bump the dates to better align with other
MOOSE changes.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1704742/) on dce2d04 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1704742/) on dce2d04 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1704742/) on dce2d04 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
Bump hashes

As a last step, bump the hashes involved.

Closes #24661

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1704789/) on a676949 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1704789/) on a676949 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1704789/) on a676949 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

cticenhour commented on conda/moose-dev/conda_build_config.yaml:
```diff
@@ -2,7 +2,10 @@ moose_libmesh:
 
 moose_wasp:
-  - moose-wasp 2023.05.01
+  - moose-wasp 2023.07.17
```
```suggestion
  - moose-wasp 2023.08.17
```

cticenhour commented on conda/moose-dev/conda_build_config.yaml:
```diff
@@ -2,7 +2,10 @@ moose_libmesh:
 
 moose_tools:
-  - moose-tools 2023.05.12
+  - moose-tools 2023.07.17
```
```suggestion
  - moose-tools 2023.08.17
```

cticenhour commented on conda/moose-dev/conda_build_config.yaml:
```diff
@@ -2,7 +2,10 @@ moose_libmesh:
+  - moose-tools 2023.07.17
+
+moose_peacock:
+  - moose-peacock 2023.07.17
```
```suggestion
  - moose-peacock 2023.08.17
```

cticenhour commented on conda/moose/conda_build_config.yaml:
```diff
@@ -1,11 +1,5 @@
-moose_python:
- - python 3.9
+moose_dev:
+  - moose-dev 2023.08.07
```
```suggestion
  - moose-dev 2023.08.17
```

cticenhour reviewed (changes requested):
There were several inconsistencies in package version definitions causing a failure on conda build for `moose-dev`, so I've put all I saw in suggestions below.

milljm commented on conda/template/conda_build_config.yaml:
```diff
@@ -1,11 +1,5 @@
-moose_python:
- - python 3.9
+moose_dev:
+  - moose-dev 2023.07.31
```
lol, woops.

👍

Commit:
Forgot to bump the important ones!

These shouldn't require a change to versioner_hashes.

Closes #24661

cticenhour commented on framework/include/utils/SymmetricRankTwoTensor.h:
```diff
@@ -144,7 +144,7 @@ class SymmetricRankTwoTensorTempl
 public:
   /// Copy assignment operator must be defined if used
-  SymmetricRankTwoTensorTempl(const SymmetricRankTwoTensorTempl<T> & a) = default;
+  // SymmetricRankTwoTensorTempl(const SymmetricRankTwoTensorTempl<T> & a) = default;
```
Did you mean to leave this commented out, rather than removing it?

milljm replied:
I... can't remember. I'll just revert that line. This was during a time I was finding warnings with the new compilers. Since I've backed out of bumping the compilers, that work is no longer needed.

moosebuild commented:
Job [Test timings](https://civet.inl.gov/job/1705991/) on c9fd46b wanted to post the following:

View timings [here](https://mooseframework.inl.gov/docs/PRs/24949/timing)

This comment will be updated on new commits.

Commit:
Remove mistakes

I beleive the test in question can operate in install mode, and was
disabled in error.

Remove the comment addition. May have been placed in error by IDE
tabs/copy/paste/enter routines?

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1707088/) on 8bd9f7a wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1707088/) on 8bd9f7a wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1707088/) on 8bd9f7a wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

loganharbour commented:
I hate to add on to this Jason, but can you add:

```
--download-hdf5-configure-arguments="--enable-cxx"
```

to the petsc configure here?

Commit:
Add requested args

By request, adding --download-hdf5-configure-arguments="--enable-cxx"
to petsc configure

Commit:
bump hashes

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1712598/) on d18a3eb wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1712598/) on d18a3eb wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
sort env

I've been wanting to add that for years...

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1712745/) on 3dab074 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1712745/) on 3dab074 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1712745/) on 3dab074 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
Correctly apply arguments for every OS

Previously was only being apply for Macintosh arch

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1714701/) on 9b70d89 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1714701/) on 9b70d89 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1714701/) on 9b70d89 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
Fix all them quotes

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1714842/) on f13d453 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

milljm commented:
@YaqiWang @NamjaeChoi 

When I enable said option, HDF5 fails to configure:

```pre
configure:31599: checking for API tracing
configure:31627: result: no
configure:31642: checking whether a memory checking tool will be used
configure:31668: result: no
configure:31680: checking whether internal memory allocation sanity checking is used
configure:31706: result: no
configure:31816: error: --enable-cxx and --enable-parallel flags are incompatible. Use --enable-unsupported to override this error.
```

Perhaps this better belongs in a separate PR while we diagnose?

Commit:
Escape the pesky double-dash

Argparse seems to require that we escape the double dash so the
option is passed and not evaluated.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1714978/) on bc9f131 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1714978/) on bc9f131 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1714978/) on bc9f131 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
Attempt --enable-cxx using separate issue

Use a dedicated issue to enable/fight --enable-cxx into HDF5

Commit:
Bump hashes as a result of PETSc change

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1715457/) on 22ff62d wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1715457/) on 22ff62d wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1715457/) on 22ff62d wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

NamjaeChoi commented:
> @YaqiWang @NamjaeChoi
> 
> When I enable said option, HDF5 fails to configure:
> 
> ```
> configure:31599: checking for API tracing
> configure:31627: result: no
> configure:31642: checking whether a memory checking tool will be used
> configure:31668: result: no
> configure:31680: checking whether internal memory allocation sanity checking is used
> configure:31706: result: no
> configure:31816: error: --enable-cxx and --enable-parallel flags are incompatible. Use --enable-unsupported to override this error.
> ```
> 
> Perhaps this better belongs in a separate PR while we diagnose?

This is not urgent, so no problem with having it as a separate PR. Thanks.

milljm commented:
@NamjaeChoi @YaqiWang 

Because you both thumbs upped the request, I am assuming your are using said feature now. How were you able to get HDF5 built with said features? No matter what I try it appears I never end up with an HDF5 built with a CXX/Parallel combination.

NamjaeChoi commented:
> @NamjaeChoi @YaqiWang
> 
> Because you both thumbs upped the request, I am assuming your are using said feature now. How were you able to get HDF5 built with said features? No matter what I try it appears I never end up with an HDF5 built with a CXX/Parallel combination.

I didn't install anything, I just included `H5Cpp.h` and it worked, so I thought it was already available. @loganharbour's thought is that the it got installed as a dependency for some other packages in the conda environment and I happened to use it.

cticenhour commented on conda/petsc/build.sh:
```diff
@@ -92,8 +92,10 @@ make
 # FIXME: Workaround mpiexec setting O_NONBLOCK in std{in|out|err}
 # See https://github.com/conda-forge/conda-smithy/pull/337
 # See https://github.com/pmodels/mpich/pull/2755
```
```suggestion
```

In conjunction with the other change here.

cticenhour commented on conda/petsc/build.sh:
```diff
@@ -92,8 +92,10 @@ make
 # See https://github.com/pmodels/mpich/pull/2755
-make check MPIEXEC="${RECIPE_DIR}/mpiexec.sh"
-
+# Skip for non-linux platforms (testing halts on ARM for some unknown reason)
+if [ `uname` == 'linux' ]; then
+  make check MPIEXEC="${RECIPE_DIR}/mpiexec.sh"
+fi
```
```suggestion
  make check
```
Let's try simply this out - it worked great for me in #23502

NamjaeChoi commented:
> > @NamjaeChoi @YaqiWang
> > Because you both thumbs upped the request, I am assuming your are using said feature now. How were you able to get HDF5 built with said features? No matter what I try it appears I never end up with an HDF5 built with a CXX/Parallel combination.
> 
> I didn't install anything, I just included `H5Cpp.h` and it worked (in my local), so I thought it was already available. @loganharbour's thought is that it got installed as a dependency for some other packages in the MOOSE conda environment and I happened to use it. I'm not sure if I'm actually using parallel HDF5.

milljm commented on conda/peacock/meta.yaml:
```diff
@@ -1,36 +1,40 @@
 
 source:
-  path: ../peacock
+  - path: ../peacock
```
Ah, nice catch. Earlier renditions of this PR included multiples (trame).

Commit:
run PETSc checks

By redacting the MPICH update, allow the PETSc checks to commence.

Commit:
versioner.yaml to track build.sh

I am questioning why this was not already added. Its major, so we
must have had a reason. And it is clear that through the build.sh
script, major changes can occur in the resulting package.

I was about to leave it, but I see we're tracking both:

  update_and_rebuild_petsci.sh|libmesh.sh
and
  configure_petsc|libmesh.sh

So why not build.sh?

Commit:
Bump versioner_hashes.yaml

With a new file to track, obviously a bump will be necessary.
I feel this is gonna break something.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1720447/) on 4f9767d wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1720447/) on 4f9767d wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1720447/) on 4f9767d wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
Allow custom prompts

Due to a recent conflict requiring a rebase, might as well add this
change here, since it would require a bump to all the same packages.

Closes #25322

Commit:
bump versioner_hashes as a result

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1728141/) on a347dbd wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1728141/) on a347dbd wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1728141/) on a347dbd wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

cticenhour commented on apptainer/mpich.def:
```diff
@@ -84,7 +84,11 @@ EOF
     # Create a pretty prompt (ensure last item sourced)
     cat <<'EOF' > ${SINGULARITY_ROOTFS}/.singularity.d/env/99-zzz_prompt.sh
-PS1="\[\033[1;34m\][`basename ${APPTAINER_NAME:-$SINGULARITY_NAME} .sif`]\[\033[1;32m\][\w]\[\033[0m\]> "
+if [ -n "${CUSTOM_PROMPT}" ]; then
+    PS1=${CUSTOM_PROMPT}
+else
+    PS1="\[\033[1;34m\][`basename ${APPTAINER_NAME:-$SINGULARITY_NAME} .sif`]\[\033[1;32m\][\w]\[\033[0m\]> "
+fi
```
🎉
